components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -5027,6 +5042,106 @@ public static String jsFunction_isScopeExist(Context cx, Scriptable thisObj,
     /**
+     * This method is used to upload backend certificate related to endpoints.
+     *
+     * @param cx      Rhino context
+     * @param thisObj Scriptable object
+     * @param args    Passing arguments {userName, alias, endpoint, certificate}
+     * @param funObj  Function object
+     * @return : True if uploading certificate is successful. False otherwise.
+     */
+    public static int jsFunction_uploadCertificate(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws APIManagementException {
+        if ((args == null) || (args.length != 4) || !isStringValues(args)) {
+            handleException("Invalid number of arguments.");
+        }
+
+        String userName = (String) args[0];
+        String alias = (String) args[1];
+        String endpoint = (String) args[2];
+        String certificate = (String) args[3];
+
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        return apiProvider.addCertificate(userName, certificate, alias, endpoint);
+    }
+
+    /**
+     * This method is used to remove backend certificate for the given alias and endpoint.
+     *
+     * @param cx      Rhino context
+     * @param thisObj Scriptable object
+     * @param args    Passing arguments {userName, alias, endpoint}
+     * @param funObj  Function object
+     * @return : True if deleting certificate is successful. False otherwise.
+     */
+    public static int jsFunction_deleteCertificate(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws APIManagementException {
+        ResponseCode responseCode = ResponseCode.INTERNAL_SERVER_ERROR;
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        if ((args == null) || (args.length != 3) || !isStringValues(args)) {
+            handleException("Invalid number of arguments.");
+        }
+
+        String userName = (String) args[0];
+        String alias = (String) args[1];
+        String endpoint = (String) args[2];
+
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        return apiProvider.deleteCertificate(userName, alias, endpoint);
+    }
+
+    /**
+     * This method is to retrieve all the certificates belongs to the given tenant.
+     *
+     * @param cx      Rhino context
+     * @param thisObj Scriptable object
+     * @param args    Passing arguments {username}
+     * @param funObj  Function object
+     * @return A list of uploaded certificate metadata.
+     */
+    public static NativeArray jsFunction_getCertificates(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws APIManagementException {
+        NativeArray certificateMetaDataArray = new NativeArray(0);
+
+        NativeObject certificateMetaData = new NativeObject();
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        if ((args == null) || (args.length != 2) || !isStringValues(args)) {
+            log.error("Invalid arguments.");
```

Comment:
Bit more descripting error message needed. basically error message should indicate what went wrong